src/main/java/com/aws/greengrass/deployment/errorcode/DeploymentErrorCodeUtils.java

```diff
@@ -180,38 +179,88 @@ private static void collectErrorCodesFromS3Exception(Set<DeploymentErrorCode> er
-                                                                  Map<String, DeploymentErrorCode> errorContext) {
+    private static void collectErrorCodesFromServiceLoadException(Set<DeploymentErrorCode> errorCodeSet) {
         errorCodeSet.add(INSTALLED_COMPONENT_NOT_FOUND);
-        if (errorContext.containsKey(ServiceLoadException.class.getSimpleName())) {
-            errorCodeSet.add(errorContext.get(ServiceLoadException.class.getSimpleName()));
+    }
+
+    /**
+     * Assign error type to deployment request errors.
+     *
+     * @param deploymentType deployment type
+     * @return nucleus error if local deployment; cloud service error if cloud deployment
+     */
+    public static DeploymentErrorType getDeploymentRequestErrorType(Deployment.DeploymentType deploymentType) {
+        switch (deploymentType) {
+            // if a local deployment request is invalid, then it's a bug in CLI and mark it as Nucleus error
+            case LOCAL:
+                return DeploymentErrorType.NUCLEUS_ERROR;
+            // if cloud deployment, then mark it cloud service error
+            case SHADOW:
+            case IOT_JOBS:
+                return DeploymentErrorType.CLOUD_SERVICE_ERROR;
+            default:
+                return DeploymentErrorType.UNKNOWN_ERROR;
         }
     }
 
     /**
-     * Check whether a component is 1p.
+     * Check whether a service is 1p.
      *
-     * @param arn component arn from metadata
-     * @return true if it's an AWS component
-     * @throws PackageLoadingException unrecognized arn
+     * @param service        service to be checked
+     * @param componentStore a reference of componentStore
+     * @return AWS component error if account is AWS; user component error if a 12 digit account; a generic component
+     *         error type if anything wrong happened.
      */
-    public static boolean isAWSComponent(@NonNull String arn) throws PackageLoadingException {
-        if (arn.isEmpty()) {
-            throw new PackageLoadingException("Empty component arn is loaded");
+    public static DeploymentErrorType classifyComponentError(GreengrassService service, ComponentStore componentStore) {
+        // get service topic for name and version
+        Topics serviceTopics = service.getServiceConfig();
+        if (serviceTopics == null) {
+            logger.atWarn().log("Null service topic while classifying component error");
+            return DeploymentErrorType.COMPONENT_ERROR;
         }
 
-        final Matcher matcher = COMPONENT_ARN_PATTERN.matcher(arn);
-        if (matcher.matches()) {
-            String accountId = matcher.group(5);
-            return "aws".equals(accountId);
+        // load component arn from recipe metadata json on disk
+        String arnString;
+        try {
+            arnString = componentStore.getRecipeMetadata(ComponentIdentifier.fromServiceTopics(serviceTopics))
```

Comment:
if the component is locally installed and there is no arn, what happens?

Reply:
we discussed offline; if a component is locally installed, then we assume it is a user component error

Reply:
is it worth thinking about including the arn in the config store as an attribute and loading from disk only if it doesn't exist?

Reply:
We discussed offline; ideally we want to put it under component’s service topic, but the arn is only received during `resolveComponentVersion` , and the greengrass service topic does not exist until we merge config and start services.

Added a TODO to think whether we need to persist it in deployment service or some other place in config.